addons/mail_client_extension/controllers/main.py

```diff
@@ -0,0 +1,323 @@
+        partner = request.env['res.partner'].with_user(request.env.user).search([('email', '=', email)], limit=1)
+        if partner:
+            response['partner'] = {
+                'id': partner.id,
+                'name': partner.name,
+                'title': partner.function,
+                'email': partner.email,
+                'image': partner.image_128,
+                'phone': partner.phone,
+                'mobile': partner.mobile,
+                'company': {},
+                'enrichment_info': None
+            }
+            # if there is already a company for this partner, just take it without enrichment.
+            if partner.parent_id:
+                response['partner']['company'] = self._get_company_dict(partner.parent_id)
+            else:
+                company = self._find_existing_company(sender_domain)
+                if not company: # create and enrich company
+                    company, enrichment_info = self._create_company_from_iap(sender_domain)
+                    response['enrichment_info'] = enrichment_info
+                response['partner']['company'] = self._get_company_dict(company)
+        else: #no partner found
+            response['partner'] = {
+                'id': -1,
+                'name': name,
+                'email': email,
+                'company': {},
+                'enrichment_info': None
+            }
+            company = self._find_existing_company(sender_domain)
+            if not company:  # create and enrich company
+                company, enrichment_info = self._create_company_from_iap(sender_domain)
+                response['enrichment_info'] = enrichment_info
+            response['partner']['company'] = self._get_company_dict(company)
+
+        return response
+
+    @http.route('/mail_client_extension/partner/create', type="json", auth="outlook", csrf=False, cors="*")
+    def res_partner_create(self, email, name, company, **kwargs):
+        # TODO search the company again instead of relying on the one provided here?
+
+        # Create the partner if needed.
+        partner_info = {
+            'name': name,
+            'email': email,
+        }
+        if company > -1:
+            partner_info['parent_id'] = company
+        partner = request.env['res.partner'].with_user(request.env.user).create(partner_info)
+
+        response = {
+            'authorized': True,
+            'created': True,
+            'id': partner.id
+        }
+        return response
+
+    @http.route('/mail_client_extension/log_single_mail_content', type="json", auth="outlook", csrf=False, cors="*")
+    def log_single_mail_content(self, lead, message, **kw):
+        response = {
+            'authorized': True,
+            'success': True
+        }
+        crm_lead = request.env['crm.lead'].with_user(request.env.user).browse(lead)
+        try:
+            crm_lead._message_log(body=message, author_id=int(request.env.user))
```

Comment:
I'm really not sure `_message_log` is the proper method to call, as the documentation notes this is an internal method which doesn't do ACL checks.

I think this should use `message_post` as first it's not really performance-critical, and second this is not a method where 
> access rights are already granted

Otherwise it seems we allow anyone to post on any lead?

/cc @tde-banana-odoo am I incorrect in this thinking?

Also to check but I think `author_id` should be a partner (not an author), and if not provided it just fallback to the current user, so passing it explicitly is probably unnecessary.

Reply:
Indeed _message_log bypasses everything to be optimized, you are right @xmo-odoo .

Depending on the result you want, you can use message_post (which will notify followers), or do an explicit acls check and then use _message_log if you want to avoid notifying anyone.

Concerning author: it is current user's partner by default so I suppose nothing to do here indeed.

Reply:
> Concerning author: it is current user's partner by default so I suppose nothing to do here indeed.

well the thing to do here is remove the explicit parameter because it's passing the user's id as partner_id, so it's going to associate a random partner no?